app/src/main/java/nu/yona/app/api/receiver/YonaReceiver.java

```diff
@@ -61,45 +61,59 @@ public void onReceive(Context context, Intent intent)
 			case AppConstant.RESTART_VPN:
 				handleRestartVPNBroadcast(context);
 				break;
+			case PowerManager.ACTION_DEVICE_IDLE_MODE_CHANGED:
+				handleDeviceDozeMode(context);
+				break;
 			default:
 				break;
 		}
 	}
 
+	@TargetApi(Build.VERSION_CODES.O)
+	private void handleDeviceDozeMode(Context context)
+	{
+		Logger.loge("BroadCast", "ACTION_DEVICE_IDLE_MODE_CHANGED");
+		PowerManager powerManager = (PowerManager) context.getSystemService(POWER_SERVICE);
+		if (powerManager.isDeviceIdleMode())
+		{
+			if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O)
```

Comment:
Combine with the above condition, like this:
```suggestion
			if (powerManager.isDeviceIdleMode() && Build.VERSION.SDK_INT >= Build.VERSION_CODES.O)
```